compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9673,22 +9670,14 @@ private STNode parseErrorTypeDescriptor(STNode errorKeywordToken) {
     /**
      * Parse error type param node.
      * <p>
-     * error-type-param := < (detail-type-descriptor | inferred-type-descriptor) >
-     * detail-type-descriptor := type-descriptor
-     * inferred-type-descriptor := *
+     * error-type-parameter := < type-descriptor >
      * </p>
      *
      * @return Parsed node
      */
     private STNode parseErrorTypeParamsNode() {
         STNode ltToken = parseLTToken();
-        STNode parameter;
-        STToken nextToken = peek();
-        if (nextToken.kind == SyntaxKind.ASTERISK_TOKEN) {
-            parameter = consume();
-        } else {
-            parameter = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_ANGLE_BRACKETS);
-        }
+        STNode parameter = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_ANGLE_BRACKETS);
         STNode gtToken = parseGTToken();
         return STNodeFactory.createErrorTypeParamsNode(ltToken, parameter, gtToken);
     }
```

Comment:
I think we can get rid of this method. Can use `parseTypeParameter()` to parse the `type-parameter`.
Also we no longer need the `ErrorTypeParamsNode`. Can remove it from the syntax tree.